sql/core/src/main/scala/org/apache/spark/sql/util/ContinuousQueryListener.scala

```diff
@@ -65,11 +65,23 @@ object ContinuousQueryListener {
   trait Event
 
   /** Event representing the start of a query */
-  class QueryStarted private[sql](val query: ContinuousQuery) extends Event
+  class QueryStarted private[sql](val queryInfo: ContinuousQueryInfo) extends Event
 
   /** Event representing any progress updates in a query */
-  class QueryProgress private[sql](val query: ContinuousQuery) extends Event
+  class QueryProgress private[sql](val queryInfo: ContinuousQueryInfo) extends Event
 
   /** Event representing that termination of a query */
-  class QueryTerminated private[sql](val query: ContinuousQuery) extends Event
+  class QueryTerminated private[sql](val queryInfo: ContinuousQueryInfo) extends Event
 }
+
+/**
+ * :: Experimental ::
+ * A class that contains information about [[ContinuousQuery]].
+ */
+@Experimental
+case class ContinuousQueryInfo(
```

Comment:
We also probably don't want this to be a `case class`.  The issue is that if we add new parameters in the future, the `unapply` method will break binary compatibility.  The constructor should also be private.